src/SonarScanner.MSBuild.Tasks/GetAnalyzerSettings.cs

```diff
@@ -326,31 +327,59 @@ private AnalyzerSettings GetLanguageSpecificSettings(AnalysisConfig config)
 
         /// <summary>
         /// Merges and returns the supplied list of file paths. In case of duplicate
-        /// // file *names* (not full paths), the path from the primary list is used.
+        /// file *names* (not full paths) of SonarAnalyzers, the path from the sonarConfiguration list is used.
         /// </summary>
-        private string[] MergeFileLists(IEnumerable<string> primaryList, IEnumerable<string> secondaryList)
+        private string[] RemoveDuplicateSonarAnalyzerReferences(IEnumerable<string> sonarConfiguration, IEnumerable<string> userProvidedConfiguration)
```

Comment:
And the method name should also reflect the merge.

Ideas:
- `MergeAndFilterAnalyzerLists` 
- `CreateAnalyzersList`
- `MergeAnalyzersLists` - for example, `MergeAdditionalFilesLists` doesn't mention de-duplication / filtering, although it does it. We could keep the same naming pattern.

Reply:
Opted for `MergeAnalyzersLists`.